tools/cldr/cldr-to-icu/src/main/java/org/unicode/icu/tool/cldrtoicu/localedistance/DistanceTable.java

```diff
@@ -0,0 +1,554 @@
+package org.unicode.icu.tool.cldrtoicu.localedistance;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static com.ibm.icu.impl.locale.LocaleDistance.DISTANCE_SKIP_SCRIPT;
+import static com.ibm.icu.impl.locale.LocaleDistance.IX_DEF_LANG_DISTANCE;
+import static com.ibm.icu.impl.locale.LocaleDistance.IX_DEF_REGION_DISTANCE;
+import static com.ibm.icu.impl.locale.LocaleDistance.IX_DEF_SCRIPT_DISTANCE;
+import static com.ibm.icu.impl.locale.LocaleDistance.IX_LIMIT;
+import static com.ibm.icu.impl.locale.LocaleDistance.IX_MIN_REGION_DISTANCE;
+import static java.util.Arrays.asList;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.logging.Logger;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Table;
+import com.google.common.collect.TreeBasedTable;
+
+/**
+ * Represents the conceptual distance between pairs of language specifications.
+ *
+ * <p>Mappings for {@code (desired, supported)} pairs are added at one of three
+ * levels in the table; language, script and region. Distances can be resolved at
+ * any level in the table (e.g. {@code ("en","fr")}, {@code ("en_Latn","ru_Cyrl")}
+ * or {@code ("en_Latn_GB", "en_Latn_AU")}).
+ *
+ * <p>However in reality the "regions" in the table are actually "partition IDs"
+ * representing groups of regions with the same language characteristics. For more
+ * information on partitions and how they are generated, see {@link PartitionInfo}.
+ *
+ * <p>This is mentioned here because anyone debugging this code might be surprised
+ * to see values like {@code "5"} for a "region" in the code. Using the term
+ * "region" matches the conceptual level of the data and is more familiar to most
+ * people, whereas "partition ID" would probably be jarring.
+ *
+ * <p>The builder class is not resuable, and once a table is built, the builder is
+ * invalid. Furthermore, since the table data itself is mutable, care must be taken
+ * to avoid modifying either the Trie or the returned distance array.
+ *
+ * <p>Note that internally the {@code '*'} character used as a wildcard for subtags
+ * is replaced by the {@code '�'} character (a.k.a ANY), whenever a subtag is
+ * passed into the API. This is because the underlying Trie structure generated by
+ * the distance table reserves {@code '*'} for a different purpose. This difference
+ * is encapsulated within this class and the {@link Trie} class only.
+ */
+final class DistanceTable {
+    private static final Logger logger = Logger.getLogger(DistanceTable.class.getName());
+
+    // Represents a wildcard match in the data table (the equivalent of '*' in
+    // <languageMatch> locale subtag). Any incoming subtags are normalized to
+    // convert '*' to this character by the builder.
+    private static final String ANY = "�";
+
+    // Distances must be in the range [0-127] because the top bit of the byte is
```

Comment:
Let's write "... because bit 7 is used ..."

The trie stores 32-bit ints, not bytes. It's a kind of varint encoding but more complicated than in a protobuf. Smaller value = smaller storage. But "of the byte" is misleading.

Reply:
If you say so. seemed pretty clear to me.

From what I can see (github being confusing and all) this was the only comment since last night, and I think everything else was resolved (I did the 2 x trieSpan things just now and everything else was done as far as I can see).
If that's the case then (given that I'll fix this) feel free to merge the PR yourself after a final check (if you can).